### PR TITLE
[CA-1048] Fix NIH relinking by adding compatibility with new Shibboleth

### PR DESCRIPTION
**Pull request by marctalbott**
Ticket: [CA-1048](https://broadworkbench.atlassian.net/browse/CA-1048)

See corresponding firecloud-develop PR: https://github.com/broadinstitute/firecloud-develop/pull/2416
and firecloud-automated-testing PR: https://github.com/broadinstitute/firecloud-automated-testing/pull/135

New Shibboleth uses a different process to sign the JWT it returns that contains the user's NIH username. We used to use a shared private key to decrypt and verify the JWT we got from Shibboleth. Now, we can use Shibboleth's public key to verify the JWT. To do so, I added a ShibbolethDAO which just grabs the public key from Shibboleth. I also decided to move all of the JWT decryption into the service level since all of the Futures played nicely together in the for comprehension and I like having less logic in the API level.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/NihApiService.scala [hunk 1]
```diff
@@ -39,38 +35,7 @@ trait NihApiService extends Directives with RequestBuilding with StandardUserInf
         path("callback") {
           post {
             entity(as[JWTWrapper]) { jwtWrapper =>
-
```
Comment (marctalbott) [inline src/main/scala/org/broadinstitute/dsde/firecloud/webservice/NihApiService.scala:42]: I moved all of the JWT decoding into NihService. I preferred having it all contained in the service and it was easier to work with all the Futures in once place

**Review by dvoet — APPROVED** [review]

**Comment by marctalbott** [comment]
Successful swatomation run with all the branches https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/44848/

**Review by s-rubenstein — APPROVED** [review]
LGTM

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/NihApiService.scala (hunk 1 above)
Comment (MatthewBemis) [reply src/main/scala/org/broadinstitute/dsde/firecloud/webservice/NihApiService.scala:42]: Agreed, thanks for moving this!

**Review by MatthewBemis — APPROVED** [review]

**Pull request merged**